### PR TITLE
Users/jnorn/myfirstlocalbranch

### PR DESCRIPTION
**Pull request by JNorn**
Updated to better class names, please see @okao21 pull request for Order Item code.

### CODE REVIEW

File: Ordering/Contracts/OrderManagement/Order.cs [hunk 1]
```diff
@@ -1,7 +1,27 @@
+using System;
+using System.Collections.Generic;
+
 
 namespace Contracts.OrderManagement
 {
     public class Order
     {
+        public int OrderNumber;
```
Comment (baez) [inline Ordering/Contracts/OrderManagement/Order.cs:9]: Please add get / Set accessors for properties you are exposing.

File: Ordering/Contracts/OrderManagement/Order.cs [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+        public string UserName;
+        public double TotalCost;
+        public DateTime DateOrdered;
+        public SortedList<Object, int> NewOrder;
+
+        public Order(int orderNumber, string userName)
+        {
+            OrderNumber = orderNumber;
+            UserName = userName;
```
Comment (baez) [inline Ordering/Contracts/OrderManagement/Order.cs:18]: Can username  be null?
If so, is it OK for it to be null?

File: Ordering/Interfaces/IOrderManager.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Ordering.Common.Interfaces
+{
+    public interface IOrderManager
+    {
+        //This is used to add an item to an order
+        void AddItem(Object Item, int Quantity);
```
Comment (baez) [inline Ordering/Interfaces/IOrderManager.cs:12]: This is an order manager so it is helpful to have a more "order specific" interface. Having an object as the input parameter here is too general.

File: Ordering/Interfaces/IOrderManager.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+        //This is used to Remove an item from an order
+        void RemoveItem(object Item);
```
Comment (baez) [inline Ordering/Interfaces/IOrderManager.cs:15]: Same as above.

**Review by baez — CHANGES_REQUESTED** [review]

**Comment by JNorn** [comment]
Changes made, and I added some extra work as well.

File: Ordering/Contracts/OrderManagement/Order.cs [hunk 5]
```diff
@@ -1,27 +1,41 @@
...
+            //Here we make sure the username isn't Null
+            try
+            {
+                if (UserName == null)
+                {
+                    throw new System.ArgumentException("Username cannot be null, no order created");
+                }
+            } catch(Exception e)
+            {
+                Console.WriteLine(e.Message);
```
Comment (baez) [inline Ordering/Contracts/OrderManagement/Order.cs]: If you want to log here you should rethrow again.

File: Ordering/Contracts/OrderManagement/Order.cs [hunk 6]
```diff
@@ -1,27 +1,41 @@
...
-        public DateTime DateOrdered;
-        public SortedList<Object, int> NewOrder;
+        public int OrderNumber { get; set; }
+        public string UserName { get; set; }
+        public double TotalCost { get; set; }
+        public DateTime DateOrdered { get; set; }
+
+        public SortedList<OrderItem, int> OrderList;
 
         public Order(int orderNumber, string userName)
```
Comment (baez) [inline Ordering/Contracts/OrderManagement/Order.cs:18]: what about order number? can it be zero?

File: Ordering/Contracts/OrderManagement/Order.cs [hunk 7]
```diff
@@ -1,27 +1,41 @@
...
-        }
+            OrderList = new SortedList<OrderItem, int>();
 
-        //Need to test order, and modify order framework to make sure it integrates
+            //Here we make sure the username isn't Null
+            try
+            {
+                if (UserName == null)
+                {
+                    throw new System.ArgumentException("Username cannot be null, no order created");
```
Comment (baez) [inline Ordering/Contracts/OrderManagement/Order.cs]: throw new ArgumentNullException(..)

**Review by baez — APPROVED** [review]

**Pull request merged**